src/components/ObsDetails/ActivityTab/ActivityHeaderContainer.js

```diff
@@ -35,19 +35,21 @@ const ActivityHeaderContainer = ( {
   const [loading, setLoading] = useState( false );
   const { user } = item;
 
-  const numFlags = item.flags?.length || 0;
+  const isFlagged = item.flags?.some( flag => !flag.resolved );
 
   useEffect( ( ) => {
     const isActiveUserTheCurrentUser = async ( ) => {
       const current = await isCurrentUser( user?.login );
       setCurrentUser( current );
     };
     isActiveUserTheCurrentUser( );
+  }, [user] );
 
-    if ( numFlags > 0 ) {
+  useEffect( ( ) => {
```

Comment:
We really don't want to use `useEffects` in code unless they're absolutely necessary, since they cause extra rerenders. In this case, you've already made `isFlagged` into a boolean, so there's no need for this line `const [flagged, setFlagged] = useState( false )` and there's no need for this effect. I'd remove both and rename `isFlagged` to `flagged`.